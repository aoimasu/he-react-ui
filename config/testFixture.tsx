import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
