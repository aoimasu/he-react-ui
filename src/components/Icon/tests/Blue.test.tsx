import fixture from '../fixtures/Blue.fixture';
import testFixture from '../../../../config/testFixture';

testFixture(fixture, 'Blue');
