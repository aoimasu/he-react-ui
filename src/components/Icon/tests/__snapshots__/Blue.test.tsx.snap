// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> rendered correctly with Blue fixture 1`] = `
<Icon
  color="blue"
  name="Cross"
  shadow={false}
>
  <Cross
    className="icon blue"
    fill="currentColor"
    height={32}
    width={32}
  />
</Icon>
`;
